main/src/rootls.cxx

```diff
@@ -0,0 +1,743 @@
+/// \param nodesEnd The last node to be printed
+/// \param flags A bitmask of RootLsArgs::Flags that influence how stuff is printed
+/// \param indent Each line of the output will have these many leading whitespaces
+static void PrintNodesDetailed(std::ostream &stream, const RootLsTree &tree,
+                               std::vector<NodeIdx>::const_iterator nodesBegin,
+                               std::vector<NodeIdx>::const_iterator nodesEnd, std::uint32_t flags, Indent indent)
+{
+   std::size_t maxClassLen = 0, maxNameLen = 0;
+   for (auto childIt = nodesBegin; childIt != nodesEnd; ++childIt) {
+      const auto &child = tree.fNodes[*childIt];
+      maxClassLen = std::max(maxClassLen, child.fClassName.length());
+      maxNameLen = std::max(maxNameLen, child.fName.length());
+   }
+   maxClassLen += 2;
+   maxNameLen += 2;
+
+   for (auto childIt = nodesBegin; childIt != nodesEnd; ++childIt) {
+      NodeIdx childIdx = *childIt;
+      const auto &child = tree.fNodes[childIdx];
+
+      const char *cycleStr = "";
+      // If this key is the first one in the list, or if it has a different name than the previous one, it means that
+      // it's the first object of that kind in the list.
+      if (childIt == nodesBegin || child.fName != tree.fNodes[childIdx - 1].fName) {
+         // Then we check the following key. If the current key is not the last key in the list and if the following key
+         // has the same name, then it means it's another cycle of the same object. Thus, it's gonna be a backup cycle
+         // of the same object. Otherwise, it's just a key with one cycle so we don't need to print information two
+         // distinguish between different cycles of the same key.
+         if (std::next(childIt) != nodesEnd && child.fName == tree.fNodes[childIdx + 1].fName) {
+            cycleStr = "[current cycle]";
+         }
+      } else {
+         // This key is a subsequent cycle of a previous key
+         cycleStr = "[backup cycle]";
+      }
+
+      PrintIndent(stream, indent);
+      stream << std::left;
+      stream << Color(kAnsiBold) << std::setw(maxClassLen) << child.fClassName << Color(kAnsiNone);
+      PrintDatime(stream, child.fKey->GetDatime());
+      std::string namecycle = child.fName + ';' + std::to_string(child.fKey->GetCycle());
+      stream << std::left << std::setw(maxNameLen) << namecycle;
+      stream << " \"" << child.fKey->GetTitle() << "\" " << cycleStr;
+      stream << '\n';
+
+      if (flags & RootLsArgs::kTreeListing) {
+         if (ClassInheritsFrom(child.fClassName.c_str(), "TTree")) {
+            TTree *tree = child.fKey->ReadObject<TTree>();
+            if (tree) {
+               PrintTTree(stream, *tree, Indent(indent + 2));
+               PrintClusters(stream, *tree, Indent(indent + 2));
+            }
+         }
+      }
+      if ((flags & RootLsArgs::kRecursiveListing) && ClassInheritsFrom(child.fClassName.c_str(), "TDirectory")) {
+         // @Recursion
```

Comment:
Leftover comment?

Reply:
I'm not against it, it just looks a bit incomplete or just placed randomly. If you're using it as a reminder of the places that introduce recursion, I guess it would be nice to give it more context. For example, by writing some doc somewhere in this file and reference it via some kind of tag in every place.